gtwrap/template_instantiator.py

```diff
@@ -201,11 +223,14 @@ class InstantiationHelper:
                   parent=parent)
     ```
     """
-    def __init__(self, instantiation_type):
+    def __init__(self, instantiation_type: InstantiatedMembers):
         self.instantiation_type = instantiation_type
 
-    def instantiate(self, instantiated_methods, method, typenames,
-                    class_instantiations, method_instantiations, parent):
+    def instantiate(self, instantiated_methods: List[InstantiatedMembers],
+                    method: ClassMembers, typenames: Sequence[str],
+                    class_instantiations: Sequence[parser.Typename],
+                    method_instantiations: Sequence[parser.Typename],
+                    parent: 'InstantiatedClass'):
```

Comment:
Re: previous comment in unit test

In this function, why isn't `method_instantiations` the same as `method.template.instantiations` and `class_instantiations` the same as `parent.template.instantiations` ?

Reply:
*Edit: Ok I see what's going on now

Reply:
Yeah I realized I was copy pasting the same pattern again and again so I just abstracted it out and tested it to make it iron clad 😁